vault/token_store.go

```diff
@@ -2205,6 +2214,87 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
 			logical.ErrInvalidRequest
 	}
 
+	// Verify the entity alias
+	overwriteEntityID := ""
+	if data.EntityAlias != "" {
+		// Parameter is only allowed in combination with token role
+		if role == nil {
+			return logical.ErrorResponse("'entity_alias' is only allowed in combination with token role"), logical.ErrInvalidRequest
+		}
+
+		// Check if provided entity alias name is in the allowed entity aliases list
+		match := false
+		for _, allowedAlias := range role.AllowedEntityAliases {
+			// Check if there is a match
+			if strings.EqualFold(allowedAlias, data.EntityAlias) {
+				match = true
+				break
+			}
+
+			// Only continue when this allowed alias is a glob pattern
+			if !strings.Contains(allowedAlias, "*") {
+				continue
+			}
+
+			// Split by asterisk to get prefix and suffix
+			split := strings.Split(allowedAlias, "*")
+
+			// Multiple asterisks are not allowed. Skip this invalid glob pattern.
+			if len(split) != 2 {
+				continue
+			}
+
+			// Check if prefix matches
+			if !strings.HasPrefix(data.EntityAlias, split[0]) {
+				continue
+			}
+
+			// Check if suffix matches
+			if !strings.HasSuffix(data.EntityAlias, split[1]) {
+				continue
+			}
+
+			// Found a match
+			match = true
```

Comment:
The algorithm for glob matching above does look very clean and sane, and has great test coverage. I did also notice that a lot of locations in Vault tend to use "https://github.com/ryanuber/go-glob" and I was wondering if we should here?

Reply:
That makes sense. I changed to `go-glob`. Thanks!